odl/trafos/fourier.py

```diff
@@ -1645,6 +1647,211 @@ def inverse(self):
             sign=sign, tmp_r=self._tmp_r, tmp_f=self._tmp_f)
 
 
+class NonUniformFourierTransformBase(Operator):
+    """Non uniform Fast Fourier Transform.
+
+    The normalization is inspired from pysap-mri, mainly this class:
+    https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/reconstruct/fourier.py#L123
+    """
+    def __init__(self, shape, non_uniform_samples, domain, range):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        shape : tuple
+            The dimensions of the data whose non uniform FFT you want to
+            compute
+        non_uniform_samples : iterable
+            List of the fourier space positions where the coefficients are
+            computed. Each position must be in [-0.5; 0.5[
+        domain : `DiscreteLp`
+            Domain of the non uniform FFT or its adjoint
+        range : `DiscreteLp`
+            Range of the non uniform FFT or its adjoint
+        """
+        super(NonUniformFourierTransformBase, self).__init__(
+            domain=domain,
+            range=range,
+            linear=True,
+        )
+        self.shape = shape
+        self._check_samples(non_uniform_samples)
+        self.non_uniform_samples = non_uniform_samples
+        self.nfft = NFFT(N=shape, M=len(non_uniform_samples))
+        self.nfft.x = non_uniform_samples
+        self.nfft.precompute()
+        self.adjoint_class = None
+
+    def _check_samples(self, non_uniform_samples):
+        """Check that samples are correct
+
+        Parameters
+        ----------
+        non_uniform_samples : iterable
+            List of the fourier space positions where the coefficients are
+            computed
+
+        Raises
+        ------
+        ValueError :
+            If the non_uniform_samples is not an iterable, of 0 length or if
```

Comment:
Prefer to move these details up top and then simply have "if the input is not valid" down here.

Reply:
See comment above.